### ISSUE TITLE
StyleGuide - In TypeScript dependency injection

### ISSUE THREAD

**Issue by mgechev** [angular/angular.io#1163]
To be discussed.

// cc @johnpapa @wardbell 

---

* Use the [`@Injectable()`](https://angular.io/docs/ts/latest/api/core/Injectable-decorator.html) decorator instead of explicitly declaring the dependencies using [`@Inject(TOKEN)`](https://angular.io/docs/ts/latest/api/testing/inject-function.html).

  *Why?*: Using [`@Injectable()`](https://angular.io/docs/ts/latest/api/core/Injectable-decorator.html) the TypeScript compiler will emit the required type metadata, which makes the code using [`@Inject()`](https://angular.io/docs/ts/latest/api/testing/inject-function.html) unnecessary verbose and less readable.

**Comment by mgechev** [comment]
This rule should be more explicit because some of the DI tokens are not types and need to be declared with `@Inject`.

**Comment by zoechi** [comment]
If you use `@Inject(SomeType)` does it need the `@Injectable()` decorator as well if it has constructor parameters?

**Comment by mgechev** [comment]
No, it doesn't. If you have `SomeType` and `SomeType1` it is better to use `@Injectable` on top of the class instead of applying two times the parameter injector `@Inject` over the two constructor parameters, i.e.:

```ts
// avoid
class Bar {
  constructor(@Inject(Foo) foo: Foo, @Inject(Baz) baz: Baz) {
    //...
  }
}
```

```ts
// recommended
@Injectable()
class Bar {
  constructor(foo: Foo, baz: Baz) {
    //...
  }
}
```

**Comment by johnpapa** [comment]
@mgechev Want to take a stab at writing this one up? I think we should ad something on this, this week. Thanks!

**Comment by johnpapa** [comment]
we added this. closed with #1170
